base/src/com/google/idea/blaze/base/sync/SyncProjectTargetsHelper.java

```diff
@@ -132,11 +132,7 @@ private static boolean shouldDeriveSyncTargetsFromDirectories(ProjectViewSet vie
     }
     ImmutableList<TargetExpression> retained =
         SourceToTargetFilteringStrategy.filterTargets(targets).stream()
-            .filter(
-                t ->
-                    t.getKind() != null
-                        && t.getKind().getLanguageClasses().stream()
-                            .anyMatch(languageSettings::isLanguageActive))
+            .filter(t -> t.getKind() != null)
```

Comment:
This change is clearly incorrect. What's the proper process for getting a new language marked as `isLanguageActive` in `bazelbuild/intellij`?

Reply:
Afaik it is done in `BlazeSyncPlugin` implementations

Reply:
This turned out to be correct - by adding the `AlwaysPresentRustSyncPlugin` and the new `WorkspaceType` `RUST`, the behaviour is now in line with other optional languages:

- issues an IDE warning if the Rust plugin is not installed, which is clickable and prompts the user to install it;
- prompts the user to `enable Bazel plugin rust support` on detecting `rust_*` targets in the project, and only once that is enabled, flags those targets for syncing

As a result, the plugin now behaves properly and passes BuildKite's checks